### ISSUE TITLE
Separate master from latest releases in docker

### ISSUE THREAD

**Issue by NyanKiyoshi** [saleor/saleor#5243]
As a user, I would like to be able to fetch the latest stable version of Saleor through docker and deploy it with minimal changes, checks and such.

As a user, I would expect `saleor:latest` or `saleor:stable` to give me the latest stable release, and, by default, having development specific instructions to not be built if not needed, i.e. file watcher. 

We could maybe even have a separate dockerfile for release/production-ready builds that would have all the settings properly set as they should be in production mode, i.e. `DEBUG=False`, no `npm` dependencies, minimal image, etc.

We could expect the development version `saleor:dev` or `saleor:master` to be shipped as a development image, being `DEBUG=True`, shipping with `npm` and a fully development capable distribution as we currently do: Debian; where in `latest`/`stable`, so the production image, it could be built on Alpine Linux instead and be kept at the strict minimal (we could end up with a image of 90-100 MiB or less instead of 245 MiB).

For the tagging, `master` or `dev` should be tagged using the git commit. `stable`/`latest` should be tagged using the version number from the git tag event.

**Comment by peelar** [comment]
@NyanKiyoshi has there been any update on that? If not, can I make it a "feature request" instead?

**Comment by NyanKiyoshi** [comment]
It's not fully correct anymore:
- We were pushing git commit hash to docker hub at some point, we decided to remove this as users can simply retrieve the image's hash digest if they need a specific commit. It also causes to have too many tags which is not great
- We are no longer pushing `master` branch (now renamed to `:main`) as `:latest`, `main` branch is now `:unstable-main`

I notice two issues with the current images we have:
- `:latest` is not pointing to the latest minor release's patch version (I see it's pointing to v3.2.7 at the moment), it should be v3.6.3
- There is currently no way to pull the latest patch release of a given minor automatically (e.g. use `:3.4` instead of having to use `:3.4.16`), this could be useful for some users

The two above issues may have been caused by an assumption that only one minor version would be supported at a time. Which is not true (we support 3.1 to 3.6 at the moment)

This ticket could be closed as it's done but there are bugs.
